core/src/test/java/com/linecorp/armeria/internal/common/stream/RecoverableStreamMessageTest.java

```diff
@@ -275,4 +275,26 @@ void nonRecoverableHttpResponse() {
                 .isInstanceOf(CompletionException.class)
                 .hasCauseInstanceOf(ClosedStreamException.class);
     }
+
+    @Test
+    void shortcutRecoverableHttpResponse() {
+        final HttpResponse failedResponse = HttpResponse.ofFailure(new IllegalStateException("test exception"));
+        final HttpResponse recovered =
+            failedResponse.recover(IllegalStateException.class, cause -> HttpResponse.of("fallback"));
+        final AggregatedHttpResponse response = recovered.aggregate().join();
+        assertThat(response.headers().status()).isEqualTo(HttpStatus.OK);
+        assertThat(response.contentUtf8()).isEqualTo("fallback");
+
+        final HttpResponseWriter failedResponse2 = HttpResponse.streaming();
+        failedResponse2.write(ResponseHeaders.of(HttpStatus.OK));
```

Comment:
This is a bit confusing because the response headers from https://github.com/line/armeria/pull/4283/files#diff-fd9ecbc60a1984ec715fcbd5a174803f219c441dc043827f13e403b243edb4afR295 has the same status. 😅 
How about specifying another status? (e.g. HttpStatus.NOT_MODIFIED)

Reply:
yeah. i will fix it :)